bundles/org.openhab.binding.hdpowerview/src/main/java/org/openhab/binding/hdpowerview/internal/database/ShadeCapabilitiesDatabase.java

```diff
@@ -289,6 +289,24 @@ public boolean supportsTilt180() {
         public boolean supportsSecondaryOverlapped() {
             return supportsSecondaryOverlapped;
         }
+
+        /**
+         * Check if the Capabilities class instance uses a vane channel.
+         *
+         * @return true if the shade supports a vane channel.
+         */
+        public boolean usesVaneChannel() {
+            return supportsTiltOnClosed || supportsTiltAnywhere;
+        }
+
+        /**
+         * Check if the Capabilities class instance uses a secondary channel.
+         *
+         * @return true if the shade supports a secondary channel.
+         */
+        public boolean usesSecondaryChannel() {
+            return supportsSecondary || supportsSecondaryOverlapped;
+        }
```

Comment:
IMHO the channel logic belongs in the handler, not in the database class. The database class should simply expose attributes/capabilities. Alternatively the methods could be renamed to not mention "Channel", so it would be more abstract and only about having vane/secondary and not about whether or not to provide a channel for that.

Reply:
I originally had the code in the calling class, but I moved it to the database to make it bit faster. However I will move it back again..